packages/flutter/lib/src/widgets/layout_builder.dart

```diff
@@ -168,10 +178,24 @@ mixin RenderConstrainedLayoutBuilder<ConstraintType extends Constraints, ChildTy
     markNeedsLayout();
   }
 
+  // Whether `_callback` must be called upon next layout.
+  //
+  // Normally the callback is only called when layout constraints change.
+  // However, we must also call the callback when widget updates, e.g. after
+  // `setState`, or `didChangeDependencies`.
+  bool _mustCallCallback = true;
```

Comment:
it's a bit of an antipattern to mutate a field on another object... maybe we should have a _needsBuild field here and instead of setting it to true, have the element call markNeedsBuild on the render object, where that's a method that sets _needsBuild and calls markNeedsLayout?

Reply:
Done.